dc_make/maker.py

```diff
@@ -134,28 +135,66 @@ def getShape(self, process, era, channel, category, model_params, unc_name=None,
         if hist is None:
           raise RuntimeError("Cannot create asimov data histogram")
       else:
-        
+        file = self.getInputFile(era, model_params)
         hist_name = f"{channel}/{category}/{process.hist_name}"
-        if unc_name is not None:
-          hist_name = f"{hist_name}_{unc_name}{unc_scale}"
-        hist = file.Get(hist_name)
-        if hist == None:
-          raise RuntimeError(f"Cannot find histogram {hist_name} in {file.GetName()}")
-      hist = self.hist_binner.applyBinning(era, channel, category, model_params, hist)
-      
-      hist.SetDirectory(0)
-      if process.scale != 1:
-        hist.Scale(process.scale)
-      if hasNegativeBins(hist):
-        axis = hist.GetXaxis()
-        bins_edges = [ str(axis.GetBinLowEdge(n)) for n in range(1, axis.GetNbins() + 2)]
-        bin_values = [ str(hist.GetBinContent(n)) for n in range(1, axis.GetNbins() + 1)]
-        print(f'bins_edges: [ {", ".join(bins_edges)} ]')
-        print(f'bin_values: [ {", ".join(bin_values)} ]')
-        raise RuntimeError(f"Negative bins found in histogram {hist_name}")
+        hists = []
+        if process.subprocesses:
+          for subp in process.subprocesses:
+            hist_name = f"{channel}/{category}/{subp}"
+            if unc_name and unc_scale:
+              hist_name += f"_{unc_name}{unc_scale}"
+            #print(channel, category, subp, unc_name, unc_scale)
+            hists.append(self.prepareHist(file,hist_name,process,listToVector(self.hist_bins[category],'double')))
+        else:
+          #print(channel, category, process, unc_name, unc_scale)
+          hists.append(self.prepareHist(file,hist_name,process, listToVector(self.hist_bins[category],'double')))
+        hist = hists[0]
+        if len(hists)>1:
+          objsToMerge = ROOT.TList()
+          for histy in hists:
+            objsToMerge.Add(histy)
+          hist.Merge(objsToMerge)
```

Comment:
```suggestion
        for sub_hist in hists[1:]:
          hist.Add(sub_hist)
```

Reply:
why not merge??

Reply:
why not add? :) We are adding histograms together, so Add is a more natural interface for this task.